src/rust/engine/src/nodes/scandir.rs

```diff
@@ -20,17 +21,40 @@ pub struct Scandir {
     pub(super) dir: Dir,
     pub(super) subject_path: SubjectPath,
 }
-
 impl Scandir {
     pub(super) async fn run_node(self, context: Context) -> NodeResult<Arc<DirectoryListing>> {
+        let gitignore_stack = self.parent_gitignore_stack(&context).await?;
         let directory_listing = context
             .core
             .vfs
-            .scandir(self.dir)
+            .scandir(self.dir, gitignore_stack)
```

Comment:
This is tricky to grok, but why is the right gitignore stack to pass here the one at the parent dir of the scanned dir, and not the one at the scanned dir itself?

Reply:
OK, still not sitting well in my head, but I think I get it...